spec/code-snippets/DeclarationReferences.ts

```diff
@@ -0,0 +1,443 @@
+  /**
+   * Shortest name:  {@link Enum12.member13}
+   * Full name:      {@link Enum12[enum].member13}
+   */
+  member13
+}
+
+// (MUST NOT have TSDoc, because this augments an already documented enum)
+export const enum Enum12 {
+  /**
+   * Shortest name:  {@link Enum12.member14}
+   * Full name:      {@link Enum12[enum].member14}
+   */
+  member14 = 14
+}
+
+
+//---------------------------------------------------------
+// Malformed names
+
+/**
+ * Shortest name:  {@link Class15}
+ * Full name:      {@link Class15[class]}
+ */
+export class Class15 {
+  /**
+   * Shortest name:  {@link Class15."abc. def"}
+   * Full name:      {@link Class15[class]."abc. def"[static]}
+   */
+  public static 'abc. def': string = 'static member with malformed characters';
+
+  /**
+   * Shortest name:  {@link Class15."abc. def"}
+   * Full name:      {@link Class15[class]."abc. def"[instance]}
+   */
+  public 'abc. def': string = 'instance member with malformed characters';
+
+  /**
+   * Shortest name:  {@link Class15.static}
+   * Also valid:     {@link Class15."static"}
+   * Full name:      {@link Class15[class].static[static]}
+   */
+  public static static: string = 'static member using keyword as name';
+
+  /**
+   * Shortest name:  {@link Class15."\uD842\uDFB7"}
+   * Full name:      {@link Class15[class]."\uD842\uDFB7"[instance]}
+   *
+   * NOTE: The string in double quotes is parsed using JSON.parse(), which converts
+   * this surrogate pair expression to the corresponding Unicode character.
+   */
+  public '𠮷': string = 'instance member using JSON unicode escapes';
+
+  /**
+   * Shortest name:  {@link Class15."\\\""}
+   * Full name:      {@link Class15[class]."\\\""[instance]}
+   *
+   * Again, the string `"\\\""` is passed to JSON.parse().
+   */
+  public '\\"': string = 'instance member using JSON escapes';
+
+  /**
+   * Shortest name:  {@link Class15.\{\}}
+   * JSON approach:  {@link Class15."\u007B\u007D"}
+   * Full name:      {@link Class15[class].\{\}[instance]}
+   *
+   * NOTE: The closing curly brace is problematic because it is a TSDoc inline tag delimiter.
+   * The "shortest name" solves this problem using TSDoc backslash escapes.  The "JSON approach"
+   * instead uses JSON unicode escapes.  Both are valid, although the JSON approach is more likely
+   * to be compatible with documentation parsers that are not TSDoc compliant.
+   */
+  public '{}': string = 'instance member using TSDoc delimiters';
+
+  /**
+   * Shortest name:  {@link Class15."1.5"}
+   * Full name:      {@link Class15[class]."1.5"[instance]}
+   *
+   * Note that the actual JavaScript object key will become a string, so "1.5" is a correct
+   * way to reference this item.
+   */
+  public 1.5: string = "a number as the key";
+}
+
+//---------------------------------------------------------
+// Generic parameters are not part of the notation
+
+/**
+ * Shortest name:  {@link Type16}
+ * Full name:      {@link Type16[type]}
+ *
```

Comment:
It might be interesting to have an example like this:

```
/**
 * @type T - {@inheritdoc Y[class].X[type]}
 */
export class X<T> extends Y<string, X> {
}
```

Reply:
If we want to support this in the future, I think it can be incorporated without any significant changes to the notation.  It seems like a fairly rare case, however, so in the interests of time I'm going to skip it for now.

---
In reply to: [218623575](https://github.com/Microsoft/tsdoc/pull/64#discussion_r218623575) [](ancestors = 218623575)